core/modules/memman/MemFile.cc

```diff
@@ -107,6 +107,21 @@ MemFile::MLResult MemFile::memLock() {
         return nilResult;
     }
 
+    // TODO: Find a better solution for systems where mmap is not viable, which
+    // manifests as no bytes being locked but bytes are reserved.
+    // Fake mmap as the configuration isn't working properly.
+    // At this point, there is enough free space, it was not a flex table,
+    // but mmap failed. So,this will fake it being a flexilock file and
+    // reserve space for it.
+    if (_memInfo.size() < freeBytes) {
```

Comment:
I can't say I am happy with the change to this logic. It really violates what MUSTLOCK means and makes for rather strange side-effects (at least they are strange to me). For instance, subsequent FLEXIBLE locks on this file are effectively upgraded to MUSTLOCK status. It also means there is no way you can really tell that a MUSTLOCK really locked anything at all.